File: Vector/OOPs/OOPsConceptAssignment/Polymorphism/InterfaceImplementation.java
// Interface Implementation in java
interface Movable {
    void move();
}

// Car.java (Class implementing Movable interface)
class Car implements Movable {
    @Override
    public void move() {
        System.out.println("Car is moving on the road.");
    }
}

// Bicycle.java (Class implementing Movable interface)
class Bicycle implements Movable {
    @Override
    public void move() {
        System.out.println("Bicycle is moving on the path.");
    }
}

// Main.java (Main class to demonstrate interface implementation)
class Main {
    public static void main(String[] args) {
        Movable car = new Car();
        Movable bicycle = new Bicycle();

        car.move();
        bicycle.move();
    }
}

//Car is moving on the road.
//Bicycle is moving on the path.

File: Vector/OOPs/OOPsConceptAssignment/Polymorphism/InterfaceImplementation.js
// Interface Implementation

//JavaScript does not have interfaces, but you can use classes to achieve similar functionality

// Define an interface Movable with a method move(). 
// Implement this interface in classes Car and Bicycle.
// Write a main method to create objects of both classes and call their move() methods.




class Movable {
    move() {
        throw new Error("This method should be overridden");
    }
}

class Car extends Movable {
    move() {
        console.log("Car is moving");
    }
}

class Bicycle extends Movable {
    move() {
        console.log("Bicycle is moving");
    }
}

function main() {
    const car = new Car();
    const bicycle = new Bicycle();

    car.move();
    bicycle.move();
}

main();

// Car is moving
// Bicycle is moving

